services/web/server/tests/unit/with_dbs/03/garbage_collector/test_resource_manager.py

```diff
@@ -564,13 +615,14 @@ async def test_interactive_services_remain_after_websocket_reconnection_from_2_t
     mocked_director_v2_api["director_v2_api.stop_dynamic_service"].assert_not_called()
     # now really disconnect
     await sio2.disconnect()
+    await sio2.wait()
     assert not sio2.sid
     # run the garbage collector
     # event after waiting some time
     await asyncio.sleep(SERVICE_DELETION_DELAY + 1)
     await garbage_collector_core.collect_garbage(client.app)
 
-    await asyncio.sleep(1)
+    await asyncio.sleep(0)
```

Comment:
why have a sleep of 0 seconds here?

Reply:
this ensures a context switch in the asyncio event loop